pkg/core/fetch.go

```diff
@@ -188,15 +188,21 @@ type FetchOptions struct {
 	ExtraFields map[string]interface{}
 	// Optional: whether fetch is executed in history mode or not
 	History *history.Config
+	// Optional: unique identifier for the fetch, if this isn't given, a random one is generated.
+	FetchId uuid.UUID
 }
 
 func Fetch(ctx context.Context, storage database.Storage, pm *plugin.Manager, opts *FetchOptions) (res *FetchResponse, diagnostics diag.Diagnostics) {
-	fetchId, err := uuid.NewUUID()
-	if err != nil {
-		return nil, diag.FromError(err, diag.INTERNAL)
+	var err error
+	fetchId := opts.FetchId
+	if fetchId == uuid.Nil {
+		fetchId, err = uuid.NewUUID()
+		if err != nil {
+			return nil, diag.FromError(err, diag.INTERNAL)
+		}
 	}
 	// set metadata we want to pass to
-	metadata := map[string]interface{}{"cq_fetch_id": fetchId.String()}
+	metadata := map[string]interface{}{"cq_fetch_id": fetchId}
```

Comment:
booo `fetchId.String()`

Reply:
Why is this an issue?

Reply:
because https://github.com/cloudquery/cq-provider-sdk/blob/66f762ee64ed563ae6646d337493b24b2abc5c7a/provider/schema/meta.go#L35